src/main/java/com/cloudbees/hudson/plugins/folder/relocate/StandardHandler.java

```diff
@@ -73,7 +77,54 @@ public HandlingMode applicability(Item item) {
         return Items.move((I) item, destination);
     }
 
-    @Override public List<? extends ItemGroup<?>> validDestinations(Item item) {
+    public boolean hasValidDestination(Item item) {
+        Jenkins instance = Jenkins.getActiveInstance();
+        if (permitted(item, instance) && instance.getItem(item.getName()) == null) {
+            // we can move to the root if there is none with the same name.
+            return true;
+        }
+        ITEM: for (Item g : instance.getAllItems()) {
```

Comment:
Ideally we need an iterator-alike method in order to avoid permission checks of all items just to exit on the first "true"